compiler/src/codegen/compcore.re

```diff
@@ -2846,7 +2862,8 @@ let compile_prim2 = (wasm_mod, env: codegen_env, p2, arg1, arg2): Expression.t =
   | LessEq
   | Int64Lte =>
     failwith("Unreachable case; should never get here: LessEq/Int64Lte")
-  | Eq =>
+  | Eq => call_equal(wasm_mod, env, [compiled_arg1(), compiled_arg2()])
```

Comment:
This does something? I thought pervasives exports the function binding, not the primitive.

Reply:
You're correct, but we can't do like `call("==", arg1, arg2)` because if someone shadowed `==`, pattern matching would break 😂 

So doing it this way guarantees that we always call the real equals function rather than what `==` is bound to.

Reply:
(Because the desugaring does `Primitive2(Eq, generated_var_pat, constant)`.)